### ISSUE TITLE
Gradient normalization in HER MPI setting

### ISSUE THREAD

**Issue by pklink1337** [openai/baselines#687]
Hi everyone,

I'm currently running some comparisons of various algorithms on the OpenAI Gym Reacher-Task and since it is so popular, HER is one of the algorithms that I would like to include in the comparison. By getting familiar with the code, I realized that in the ddpg.py file in the 'her' subfolder, the MpiAdam optimizers for the Q- and Policy-Network get initialized via:

```
        self.Q_adam = MpiAdam(self._vars('main/Q'), scale_grad_by_procs=False)
        self.pi_adam = MpiAdam(self._vars('main/pi'), scale_grad_by_procs=False)
```

In [this issue](https://github.com/openai/baselines/issues/314) , it was stated that the gradients computed by the individual MPI workers get averaged before being applied. 

Now I was wondering where in the code that happens, since the "scale_grad_by_procs" flag would actually accomplish this, but it is specifically disabled. 

I'm that interested in it because I want to make the experiment results independent of the number of CPUs used by changing the options 'batch_size' and 'rollout_batch_size' according to the number of available CPUs, so that the total batch size and the total number of rollouts per cycle are the same as in the 19 Core experimental setting.

Now if the gradients would not get normalized, the magnitude of the gradient would also scale linearly with the number of available CPUs, so I would need to take that into account as well.

So if anyone could help me out here, I would be very happy.

Best
Pascal

**Comment by pzhokhov** [comment]
I am not very intimately familiar with HER implementation yet, so inviting @machinaut and @matthiasplappert to comment on this; however, my understanding is that we indeed are not averaging, but summing the gradients from MPI workers by virtue of `scale_grad_by_procs=False` flag (so if you'd like a fair comparison when varying number of workers, `scale_grad_by_procs=True` may be a better idea . However, that may require adjustment of some other hyperparameters if you want to nonetheless reproduce the results from HER paper exactly).

**Comment by matthiasplappert** [comment]
This is correct, the implementation is summing and not averaging. You can try to use averaging instead. That being said, since we use the Adam optimizer, this shouldn't matter much since it will get absorbed into Adam's normalization of gradients anyway.
